fairlearn/datasets/_fetch_acs_public_coverage.py

```diff
@@ -0,0 +1,133 @@
+# Copyright (c) Microsoft Corporation and Fairlearn contributors.
+# Licensed under the MIT License.
+
+import pathlib
+
+from sklearn.datasets import fetch_openml
+from ._constants import _DOWNLOAD_DIRECTORY_NAME
+from ._utils import _STATE_CODES, check_states_valid
+
+
+def fetch_acs_public_coverage(
+    *,
+    cache=True,
+    data_home=None,
+    as_frame=False,
+    return_X_y=False,
+    states=None,
+):
+    """Load the ACS Public Coverage dataset.
+
+    Read more in the :ref:`User Guide <acs_public_coverage>`.
+
+    Download it if necessary.
+
+    ==============   ==============
+    Samples total         1,138,289
+    Dimensionality               19
+    Features                   real
+    Target                     real
+    ==============   ==============
+
+    Source: Paper: Frances Ding et al. (2021) [1]_
+            and corresponding repository https://github.com/zykls/folktables/
+
+    .. versionadded:: 0.7.1
+
+    Prediction task is to determine whether a person has public
+    health coverage.
+
+    Parameters
+    ----------
+    cache : bool, default=True
+        Whether to cache downloaded datasets using joblib.
+
+    data_home : str, default=None
+        Specify another download and cache folder for the datasets.
+        By default, all fairlearn data is stored in '~/.fairlearn-data'
+        subfolders.
+
+    as_frame : bool, default=False
+        If True, the data is a pandas DataFrame including columns with
+        appropriate dtypes (numeric, string or categorical). The target is
+        a pandas DataFrame or Series depending on the number of target_columns.
+        The Bunch will contain a ``frame`` attribute with the target and the
+        data. If ``return_X_y`` is True, then ``(data, target)`` will be pandas
+        DataFrames or Series as describe above.
+
+    return_X_y : bool, default=False
+        If True, returns ``(data.data, data.target)`` instead of a Bunch
+        object.
+
+    states : list, default=None
```

Comment:
I am not sure whether we need to provide this functionality in Fairlearn. It's quite trivial to filter the data set after it is fetched.

Reply:
if it was a single liner I'd agree with you, but I'm happy to keep this since it's a bit more than that.

Reply:
Fair enough. In that case let's leave it in :)